controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -71,10 +59,26 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return ctrl.Result{}, nil
 	}
 
-	// check that there is a ToolchainCluster CR in the same namespace
+	readyCondition, reportedError := r.determineReadyState(ctx, spaceProvisionerConfig)
+
+	spaceProvisionerConfig.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(spaceProvisionerConfig.Status.Conditions,
+		readyCondition)
```

Comment:
I know that it doesn't matter much in this case, but you could do:
```suggestion
	var updated bool
	spaceProvisionerConfig.Status.Conditions, updated = condition.AddOrUpdateStatusConditions(spaceProvisionerConfig.Status.Conditions,
		readyCondition)
	if !updated {
		return ctrl.Result{}, reportedError
	}
```
but not probably needed applying it in this case :-)